cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/realtime/sink/RealtimeElasticsearchSink.java

```diff
@@ -0,0 +1,161 @@
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.templates.plugins.PluginConfig;
+import co.cask.cdap.template.etl.api.realtime.DataWriter;
+import co.cask.cdap.template.etl.api.realtime.RealtimeContext;
+import co.cask.cdap.template.etl.api.realtime.RealtimeSink;
+import co.cask.cdap.template.etl.common.Properties;
+import co.cask.cdap.template.etl.common.StructuredRecordStringConverter;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import org.elasticsearch.action.bulk.BulkItemResponse;
+import org.elasticsearch.action.bulk.BulkRequestBuilder;
+import org.elasticsearch.action.bulk.BulkResponse;
+import org.elasticsearch.client.transport.TransportClient;
+import org.elasticsearch.common.settings.ImmutableSettings;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.transport.InetSocketTransportAddress;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+
+/**
+ * Realtime sink for elasticsearch.
+ */
+@Plugin(type = "sink")
+@Name("Elasticsearch")
+@Description("Real-time sink that outputs to the Elasticsearch server.")
+public class RealtimeElasticsearchSink extends RealtimeSink<StructuredRecord> {
+  private static final Logger LOG = LoggerFactory.getLogger(RealtimeElasticsearchSink.class);
+  private static final String INDEX_DESCRIPTION = "The name of the index where the data will be stored. " +
+    "If the index does not already exist, it will be created using Elasticsearch's default properties.";
+  private static final String TYPE_DESCRIPTION = "The name of the type where the data will be stored. " +
+    "If it does not already exist, it will be created.";
+  private static final String ID_DESCRIPTION = "The field that will determine the id for the document. " +
+    "It should match a field name in the structured record of the input.";
+  private static final String TRANSPORT_ADDRESS_DESCRIPTION = "The addresses for nodes. " +
+    "Specify the address for at least one node, and separate others by commas. Other nodes will be sniffed out. " +
+    "For example: host1:9300,host2:9300.";
+  private static final String CLUSTER_DESCRIPTION = "The name of the cluster to connect to. " +
+    "Defaults to \'elasticsearch\'.";
+
+  private final RealtimeESSinkConfig realtimeESSinkConfig;
+  private TransportClient client;
+
+  public RealtimeElasticsearchSink(RealtimeESSinkConfig realtimeESSinkConfig) {
+    this.realtimeESSinkConfig = realtimeESSinkConfig;
+  }
+
+  @Override
+  public void initialize(RealtimeContext context) {
+    realtimeESSinkConfig.cluster = Strings.isNullOrEmpty(realtimeESSinkConfig.cluster) ?
+      "elasticsearch" : realtimeESSinkConfig.cluster;
+
+    Settings settings = ImmutableSettings.settingsBuilder()
+      .put("node.name", "cdap")
```

Comment:
Will it be alright to set to a constant value? Will it still work if we have two or more adapters with RealtimeESSink?